### ISSUE TITLE
 How to incorporate loss function into my model？

### ISSUE THREAD

**Issue by 202041600047** [tfzhou/ContrastiveSeg#45]
Thank you very much for your work, I have a question: my model is BiSeNetV2, I want to replace my loss function with your pixel contrast loss function, but my output has only one value, and the preds of the pixel contrast loss function is a dictionary, and the keys are 'seg' and 'embed', I don't know what 'seg' and 'embed' mean, and I don't know how to get these two values ​​in BiSeNetV2?

class ContrastCELoss(nn.Module, ABC):
    def forward(self, preds, target, with_embed=False):
        h, w = target.size(1), target.size(2)
        assert "seg" in preds
        assert "embed" in preds
        seg = preds['seg']
        embedding = preds['embed']

**Comment by tfzhou** [comment]
@202041600047 Thanks for your interests.  For the `seg`, it is the output of the model; while for the `embed`, you can obtain it through a convolutional projection head of your backbone model. Sorry for that I am not very familiar with the structure of BiSeNetV2, and cannot tell the specific layer that you can add the projection head.

**Comment by 202041600047** [comment]
> @202041600047 Thanks for your interests. For the `seg`, it is the output of the model; while for the `embed`, you can obtain it through a convolutional projection head of your backbone model. Sorry for that I am not very familiar with the structure of BiSeNetV2, and cannot tell the specific layer that you can add the projection head.

Thank you very much for answering my question in your busy schedule. I don't understand the project head very much. Where is the project head in the code and what is its function? Thank you very much for answering my question！

**Comment by heifanfanfan** [comment]
> 

Have you solved the problem of loading the new model? I have the same problem
